lib/vpago/vattanac_mini_app/base.rb

```diff
@@ -41,11 +40,11 @@ def expired_at
       end
 
       def partner_code
-        ENV['VATTANAC_PARTNER_CODE'].presence
+        @payment.payment_method.preferred_partner_code
       end
 
       def refund_url
-        ENV['VATTANAC_REFUND_URL'].presence
+        ENV['VATTANAC_REFUND_URL'].presence || Rails.application.credentials.vattanac.refund_url
```

Comment:
Let’s move all the environment variables to the preference config and update the README with sample values and setup instructions for each one.